File: client/sources.go
package client

import (
	"fmt"

	"github.com/unluckythoughts/manga-reader/server/models"
)

// ListSources retrieves a paginated list of sources
func (c *Client) ListSources(page, limit int) (*models.SourcesPaginatedResponse, error) {
	url := fmt.Sprintf("/api/v1/sources?page=%d&limit=%d", page, limit)

	var response models.SourcesPaginatedResponse
	_, err := c.client.GetResponse(url, &response)
	if err != nil {
		return nil, err
	}

	return &response, nil
}

// GetSource retrieves a source by ID
func (c *Client) GetSource(id int) (*models.Source, error) {
	url := fmt.Sprintf("/api/v1/sources/%d", id)

	var response models.Source
	_, err := c.client.GetResponse(url, &response)
	if err != nil {
		return nil, err
	}

	return &response, nil
}

File: app/src/api/sources.ts
import { apiClient } from './client'
import type { Source, PaginatedResponse } from '../types'

export const sourcesApi = {
  list: (page = 1, limit = 20) =>
    apiClient.get<PaginatedResponse<Source>>(`/reader/sources?page=${page}&limit=${limit}`),
  get: (id: number) => apiClient.get<Source>(`/reader/sources/${id}`)
}
